spec/rubocop/cop/rails/active_record_callbacks_order_spec.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Rails::ActiveRecordCallbacksOrder do
+  subject(:cop) { described_class.new }
+
+  it 'registers an offense and corrects when declared callbacks are not correctly ordered' do
+    expect_offense(<<~RUBY)
+      class User < ApplicationRecord
+        scope :admins, -> { where(admin: true) }
+
+        after_commit :after_commit_callback
+        after_save :after_save_callback
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `after_save` is supposed to appear before `after_commit`.
+
+        def some_method
+        end
+
+        before_validation :before_validation_callback
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `before_validation` is supposed to appear before `after_save`.
+        some_other_macros :foo
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      class User < ApplicationRecord
+        scope :admins, -> { where(admin: true) }
+
+        before_validation :before_validation_callback
+        after_save :after_save_callback
+        after_commit :after_commit_callback
+
+        def some_method
+        end
+
+        some_other_macros :foo
+      end
+    RUBY
+  end
+
+  it 'correcly autocorrects when there is a comment for callback method' do
+    new_source = autocorrect_source(<<~RUBY)
+      class User < ApplicationRecord
+        # This is a
+        # multiline
+        # comment for after_commit.
+        after_commit :after_commit_callback
+        after_save :after_save_callback
+      end
```

Comment:
This is a good test case 👍 Can you update the following cases further? I think it would be even better if the test included moving comments.

```suggestion
      class User < ApplicationRecord
        # This is a
        # multiline
        # comment for after_commit.
        after_commit :after_commit_callback
        # This is another
        # multiline
        # comment for after_save.
        after_save :after_save_callback
      end
```